src/bp/core/services/middleware/event-collector.ts

```diff
@@ -87,9 +104,16 @@ export class EventCollector {
       workflowId: activeWorkflow?.eventId,
       success: activeWorkflow?.success,
       incomingEventId: event.direction === 'outgoing' ? incomingEventId : id,
-      event: this.knex.json.set(ignoredProps.length ? _.omit(event, ignoredProps) : event || {}),
+      event: ignoredProps.length ? (_.omit(event, ignoredProps) as sdk.IO.Event) : event,
       createdOn: this.knex.date.now()
-    })
+    }
+
+    const exists = this.batch.findIndex(x => x.id === id)
```

Comment:
exists should be a boolean so it may be rewritten like this:

```
const exists = this.batch.findIndex(x => x.id === id) !== -1
if (exists) {
...
```

Reply:
Needs to be a number because just below I need the index to remove the element from the array. Renamed it